### ISSUE TITLE
frac的问题

### ISSUE THREAD

**Issue by Radleye** [HAIRLAB/DeceFL#3]
您好，我想问一下frac设置的问题。代码中如果将frac设置为小于1的数，如100个客户端，frac=0.1就会显示超出下标，由于 global_model_i这个list的长度是等于Num_model = int(args.frac * args.num_users)的大小(10)。而实际在选择客户端又用的下面这个idxs_users = np.random.choice(range(args.num_users), m, replace=False)，这个idxs_users 的下标是1-100大于10的。因此本地模型在训练的时候            model_i_update, loss = local_model.update_weights(
                model=copy.deepcopy(global_model_i[idx]), global_round=epoch,idx=idx)会显示下标超出，global_model_i这个list的范围(因为idx是1-100随机的。而global_model_i只有10个)。这个是什么意思，意思frac只能设置为1吗？不能只选部分客户端吗，怎么改

**Comment by JinDou1210** [comment]
frac是fedavg历史遗留下来的参数，在我们的decefl实验中其实没有用到过它（一直保持的是1），建议不要更改frac的值。如果你想构造只选部分客户的场景，可以通过设置权重矩阵w实现。
